File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/adapter/transport/Plane.java
package com.bigboxcode.designpattern.adapter.transport;

public class Plane  implements AirTransport {
    @Override
    public int getNumberOfWheels() {
        return 3;
    }

    @Override
    public int getNumberOfEngines() {
        return 2;
    }

    // get weight in pound
    @Override
    public double getWeight() {
        return 127_000;
    }

    // In Nautical miles
    @Override
    public double getDistanceTravelled() {
        return 500; // Nautical files
    }

    @Override
    public double getTravelCostTotal() {
        return 3_000;
    }
}

File: design-patterns/php/adapter/transportadapter/Plane.php
<?php
// Plane.php

namespace BigBoxCode\DesignPattern\Adapter\TransportAdapter;


class Plane implements AirTransport {
    public function getNumberOfWheels(): int {
        return 3;
    }

    public function getNumberOfEngines(): int {
        return 2;
    }

    public function getWeight(): float {
        return 127_000;
    }

    public function getDistanceTravelled(): float {
        return 500;
    }

    public function getTravelCostTotal(): float {
        return 3_000;
    }
}